### ISSUE TITLE
Support for azure_monitor_metric_alert skipMetricValidation option in criteria block

### ISSUE THREAD

**Issue by S48-Mo** [terraform-providers/terraform-provider-azurerm#10359]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

As per https://github.com/MicrosoftDocs/azure-docs/issues/16061 there has been work to update the Azure Go SDK to provide a `skipMetricValidation` field in the alert criteria, which will provide the ability to create alerts for custom metrics that don't yet exist in Monitor. This helps with preventing race conditions where the metric hasn't emitted to Monitor before the alert is created.

Looking at the Go SDK for v.49.2.0 (which is supported by azurerm v2.44.0) it looks like field in question is definitely in the MetricCriteria type https://github.com/Azure/azure-sdk-for-go/blob/v49.2.0/services/preview/monitor/mgmt/2019-06-01/insights/models.go . It looks like it's just a matter of adding support for that field in azurerm.

### New or Affected Resource(s)

* azurerm_monitor_metric_alert

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
resource "azurerm_monitor_metric_alert" "foo" {
  name                             = "alert-foo"
  resource_group_name = "rg-foo"
  scopes = [azurerm_application_insights.app_insights.id]

  criteria {
    metric_namespace = "Microsoft.Insights/Components"
    metric_name = "custom-metric-not-yet-in-monitor"
    aggregation = "Average"
    operator = "GreaterThan"
    threshold = 0
    skipMetricValidation = true # This field will be new
  }
  frequency = "PT1H"
  window_size = "PT1H"

  action {
    action_group_id = azurerm_monitor_action_group.action_group.id
  }
}
```

### References

* https://github.com/MicrosoftDocs/azure-docs/issues/16061

**Comment by jbinko** [comment]
@tombuildsstuff, @yupwei68, @S48-Mo I think this is duplicit with: #10421. Implemented in#10422.

**Comment by S48-Mo** [comment]
Thanks @jbinko. Closing this issue.
